lib/rubocop/cop/rspec/unspecified_exception.rb

```diff
@@ -0,0 +1,83 @@
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for a specified error in checking raised errors.
+      #
+      # Enforces one of an Exception type, a string, or a regular
+      # expression to match against the exception message as a parameter
+      # to `raise_error`
+      #
+      # @example
+      #
+      #     # bad
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error
+      #
+      #     # good
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError)
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError) { |err| expect(err.data).to eq(42) }
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError) do |err|
+      #       expect(err.data).to eq(42)
+      #     end
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error('error')
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError, 'error')
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(/err/)
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError, /err/)
+      #
+      #     expect { do_something }.not_to raise_error
+      class UnspecifiedException < Cop
+        NO_PARAMETER_MSG = 'Specify the exception being captured'.freeze
+
+        EMPTY_RAISE_ERROR_PATTERN = <<-PATTERN.freeze
+          (send
+            (block
+                (send nil? :expect) ...)
+            :to
+            (send nil? :raise_error)
+          )
+        PATTERN
+        def_node_matcher :empty_raise_error_call, EMPTY_RAISE_ERROR_PATTERN
+
+        def_node_matcher :raise_error_with_block, <<-PATTERN.freeze
+          (block
+            #{EMPTY_RAISE_ERROR_PATTERN}
+            (args (arg ...))
+            ...
+          )
+        PATTERN
+
+        def on_send(node)
+          return unless empty_raise_error_call(node) &&
+            !raise_error_with_block(node.parent)
```

Comment:
What does the `raise_error_with_block` check on the parent node do?

Reply:
oh!  I think this is exactly what I was referring to before, that is the matcher that catches `do`/`end` defined blocks but without it curly-braced blocks pass!  To validate I commented out the matcher and the second half of the conditional and the `allows classes with blocks with do/end` example failed.